src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -84,8 +89,9 @@
         String[] fileDiffResultList = fullDiffResult.split(DIFF_FILE_CHUNK_SEPARATOR);
 
         for (String fileDiffResult : fileDiffResultList) {
-            // file deleted or is a binary file, skip it
-            if (fileDiffResult.startsWith(FILE_DELETED_METADATA) || fileDiffResult.contains(BINARY_FILE_SYMBOL)) {
+            // file deleted, is binary file or is new and empty file, skip it
+            if (fileDiffResult.startsWith(FILE_DELETED_METADATA) || fileDiffResult.contains(BINARY_FILE_SYMBOL)
+                    || NEW_EMPTY_FILE_PREDICATE.test(fileDiffResult)) {
```

Comment:
actually, why don't do it similarly as `FILE_DELETED_METADATA`?

Reply:
For files that are deleted,the `git diff` result will still print out all the lines in the file (as deleted)
A new file also does the same, printing out all the lines in the file (as added).
However, what we are checking here is a special case where if we add a new file, but that file is empty, thus not having the header group we are trying to find, or any lines to print as well.